### ISSUE TITLE
WAITING PYTHON 3 UPGRADE: Diacritical marks in index

### ISSUE THREAD

**Issue by MartaSztuka** [openstax/cnx-recipes#1042]
https://katalyst01.cnx.org/contents/TqqPA4io@1.98:lyD6ZIjV@1.98/Index-of-terms
https://cnx.org/contents/TqqPA4io@1.134:lyD6ZIjV@1.134/Skorowidz-rzeczowy
![fiz4pol](https://user-images.githubusercontent.com/47241114/56897890-f54f6700-6a8f-11e9-9300-635c2e91cdca.jpeg)
Index on katalyst01 is by english alphabet, polish letters are all together, before 'A'.

This issue occurs in all 3 volumes:
![fizfiz](https://user-images.githubusercontent.com/47241114/56900560-66921880-6a96-11e9-99c6-d91844ef6b1f.jpeg)

**Comment by aiwenar** [comment]
Terms starting with non-English characters being sorted into _Symbols_ is likely related to #367; To get them to sort properly one needs to set (if I remember correctly)

```sass
$Config_Index_SymbolRegexp: "^\W";
$Config_Index_NotSymbolRegexp: "^\w";
```

as by default recipes treat anything not beginning with an English character as a symbol.

I'm unsure about “Tesla, Nicola” and “isothermal expansion” though, as they do seem to begin with English characters. That might be, as @openstaxalina suggested in #1010, caused by a space preceding the term, as I'm seeing those problems on cnx.org's version of _Fizyka dla szkół wyższych_ too.

**Comment by MartaSztuka** [comment]
EDIT: Two cases, where non-english letter doesn't ocure (Tesla, Nicola and isotermal) were content issue and now are fixed.

**Comment by MartaSztuka** [comment]
All 3 books look good. Tested on katalyst01
![Screenshot from 2019-05-02 14-59-16](https://user-images.githubusercontent.com/47241114/57076921-0516b780-6ceb-11e9-904d-c5e40f43240b.png)
![Screenshot from 2019-05-02 15-28-47](https://user-images.githubusercontent.com/47241114/57078717-1366d280-6cef-11e9-9fe5-7f4feb56db73.png)

**Comment by openstaxalina** [comment]
@Adrianltd has fixed all this locally; however, when tested on the server, Symbols are still sorting to the top of the Index when they should sort to the end of the Index.  

@aiwenar pointed us to the commit that is making the regex he created no longer work: https://github.com/openstax/cnx-recipes/commit/c69fa924e99600c6ccb17bcbba9233f7cad3f26b
Lines 202/200 - quotes were removed due to a restriction of node-sass.  

There may be a different in software between the server and the local environment that is causing this.  Options:
- easybake needs an update - @tomjw64 please specify what version easybake should be.  The Houston team will ask DevOps to verify that version.
- cnx-deploy dependencies need to be updated
- cnx-epub?

**Comment by tomjw64** [comment]
@openstaxalina After looking at katalyst01's python dependencies, I can't identify any out of date libraries, which is what caused a similar issue on our servers.

**Comment by karenc** [comment]
I downloaded the book from katalyst01.cnx.org and the ruleset.css and I can confirm that locally the index is sorted correctly but on katalyst01 it is not.

**TL;DR** cnx-easybake behaves differently on the server (which runs python 2.7) and locally (which uses python 3.6), causing the sorting problem. :disappointed: We'll open an issue for this.

This is what I did locally:

```
# This looks different from the normal baked pdf CLI workflow because I didn't add the book to books.txt
# Download the book from katalyst01.cnx.org
mkdir -p data/col25571_1.3.26/raw
docker-compose run --rm neb neb get -d /out/col25571_1.3.26/raw katalyst01.cnx.org col25571 ''
# Download the recipe linked in https://katalyst01.cnx.org/a/content-status/
wget -O tom3-ruleset.css 'https://katalyst01.cnx.org/resources/c0c4cb73fb3d0db3a554b62fa738394fe549c6b1'
# Assemble the book
docker-compose run --rm neb neb assemble /out/col25571_1.3.26/raw /out/col25571_1.3.26
# Bake the book
docker-compose run --rm bake-book /bin/bash -c 'venv/bin/cnx-easybake tom3-ruleset.css data/col25571_1.3.26/collection.assembled.xhtml data/col25571_1.3.26/collection.baked.xhtml'
```

Checked the baked html in the browser and can confirm that the order is correct.

Then I did this on katalyst01.cnx.org:

```
# Upload the assembled file to katalyst01
# scp data/col25571_1.3.26/collection.baked.xhtml katalyst01.cnx.org:/tmp/
# Download the recipe on katalyst01
wget -O tom3-ruleset.css 'https://katalyst01.cnx.org/resources/c0c4cb73fb3d0db3a554b62fa738394fe549c6b1'
# Run cnx-easybake
/var/cnx/venvs/publishing/bin/cnx-easybake tom3.css tom3.assembled.xhtml tom3.baked.xhtml
```

Checked the baked html in the browser and indeed the order is wrong.

The cnx-easybake version for local and katalyst01 are both `1.2.5`.  The only difference now I see is python 2 and python 3.

As an experiment, I did this locally:

```
virtualenv -p `which python2` py2
py2/bin/pip install cnx-easybake==1.2.5
py2/bin/cnx-easybake tom3-ruleset.css data/col25571_1.3.26/collection.assembled.xhtml tom3.baked.xhtml
```

The result is same as the server.

**Comment by karenc** [comment]
By the way, if I download the [ruleset.css from staging](https://staging.cnx.org/resources/167b8ee5863e682f45c406cd8cb55fba46cca385), it works in python 2.  But still, we should fix this bug so the ruleset.css on katalyst01.cnx.org works.

**Comment by helenemccarron** [comment]
@karenc Thank you, this helps a lot! It looks like the next step is to figure out why the proposed fix doesn't work with Python 3.

**Comment by karenc** [comment]
I narrowed down the bit of the recipe that doesn't work in python 2:

```
  :pass(3) div[data-type="page"] span[data-type="term"][cmlnle|reference]::after,
  :pass(3) div[data-type="composite-page"] span[data-type="term"][cmlnle|reference]::after {
    container: span;
    data-type: term-content;
    content: attr(cmlnle|reference); }

  :pass(3) div[data-type="page"] span[data-type="term"]:not([cmlnle|reference])::after,
  :pass(3) div[data-type="composite-page"] span[data-type="term"]:not([cmlnle|reference])::after {
    container: span;
    data-type: term-content;
    content: content(); }
```

If I comment this bit out, the ordering is correct when running with python 2 and python 3.

I have no idea what it really means but does anyone have any ideas?

**Comment by tomjw64** [comment]
@karenc What did you do to isolate this broken snippet, if you don't mind me asking?

**Comment by karenc** [comment]
@tomjw64 I was trying to create a minimal recipe that only does the index of foreign terms which can demonstrate the problem (for unit testing purposes) and found that if I delete the rules from pass 0 to 5, the order is correct... I was like "???" :thinking:  So I tried to kept commenting out stuff and bake and see when the order is correct and when it is wrong.

**Comment by openstaxalina** [comment]
We need to release University Physics in baked styles at the same time as (or after) the python upgrades have happened.

**Comment by openstaxalina** [comment]
Temporary solution was to remove the problematic terms from the Index.  cc @paulinaszczucinska

**Comment by brittweinstein** [comment]
Closing for now -- we can reopen when we prioritize Python 3
